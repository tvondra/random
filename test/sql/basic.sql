CREATE EXTENSION random;

SELECT COUNT(*), COUNT(DISTINCT random_string(1, 100, 32, 64)) > 95 FROM generate_series(1,10000);

SELECT COUNT(*), COUNT(DISTINCT random_bytea(1, 100, 32, 64)) > 95 FROM generate_series(1,10000);

SELECT COUNT(*), COUNT(DISTINCT random_int(1, 100, 1, 1000000)) > 95 FROM generate_series(1,10000);

SELECT COUNT(*), COUNT(DISTINCT random_bigint(1, 100, 1, 1000000)) > 95 FROM generate_series(1,10000);

SELECT COUNT(*), COUNT(DISTINCT random_real(1, 100, 1, 1000000)) > 95 FROM generate_series(1,10000);

SELECT COUNT(*), COUNT(DISTINCT random_double_precision(1, 100, 1, 1000000)) > 95 FROM generate_series(1,10000);

SELECT COUNT(*), COUNT(DISTINCT random_numeric(1, 100, 1, 1000000)) > 95 FROM generate_series(1,10000);

SELECT COUNT(*), COUNT(DISTINCT random_macaddr(1, 100)) > 95 FROM generate_series(1,10000);

SELECT COUNT(*), COUNT(DISTINCT random_macaddr8(1, 100)) > 95 FROM generate_series(1,10000);

SELECT COUNT(*), COUNT(DISTINCT random_inet(1, 100)) > 95 FROM generate_series(1,10000);

SELECT COUNT(*), COUNT(DISTINCT random_cidr(1, 100)) > 95 FROM generate_series(1,10000);

SELECT COUNT(*), COUNT(DISTINCT random_cidr2(1, 100)) > 95 FROM generate_series(1,10000);
