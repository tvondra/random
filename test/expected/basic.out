CREATE EXTENSION random;
SELECT COUNT(*), COUNT(DISTINCT random_string(1, 100, 32, 64)) FROM generate_series(1,10000);
 count | count 
-------+-------
 10000 |   100
(1 row)

SELECT COUNT(*), COUNT(DISTINCT random_bytea(1, 100, 32, 64)) FROM generate_series(1,10000);
 count | count 
-------+-------
 10000 |   100
(1 row)

SELECT COUNT(*), COUNT(DISTINCT random_int(1, 100, 1, 1000000)) FROM generate_series(1,10000);
 count | count 
-------+-------
 10000 |   100
(1 row)

SELECT COUNT(*), COUNT(DISTINCT random_bigint(1, 100, 1, 1000000)) FROM generate_series(1,10000);
 count | count 
-------+-------
 10000 |   100
(1 row)

SELECT COUNT(*), COUNT(DISTINCT random_real(1, 100, 1, 1000000)) FROM generate_series(1,10000);
 count | count 
-------+-------
 10000 |   100
(1 row)

SELECT COUNT(*), COUNT(DISTINCT random_double_precision(1, 100, 1, 1000000)) FROM generate_series(1,10000);
 count | count 
-------+-------
 10000 |   100
(1 row)

SELECT COUNT(*), COUNT(DISTINCT random_numeric(1, 100, 1, 1000000)) FROM generate_series(1,10000);
 count | count 
-------+-------
 10000 |   100
(1 row)

SELECT COUNT(*), COUNT(DISTINCT random_macaddr(1, 100)) FROM generate_series(1,10000);
 count | count 
-------+-------
 10000 |   100
(1 row)

SELECT COUNT(*), COUNT(DISTINCT random_macaddr8(1, 100)) FROM generate_series(1,10000);
 count | count 
-------+-------
 10000 |   100
(1 row)

SELECT COUNT(*), COUNT(DISTINCT random_inet(1, 100)) FROM generate_series(1,10000);
 count | count 
-------+-------
 10000 |   100
(1 row)

SELECT COUNT(*), COUNT(DISTINCT random_cidr(1, 100)) FROM generate_series(1,10000);
 count | count 
-------+-------
 10000 |   100
(1 row)

SELECT COUNT(*), COUNT(DISTINCT random_cidr2(1, 100)) FROM generate_series(1,10000);
 count | count 
-------+-------
 10000 |   100
(1 row)

